src/components/inputs/money-input/money-input.js

```diff
@@ -328,59 +370,55 @@ export default class MoneyInput extends React.Component {
+      label: currencyCode,
+      value: currencyCode,
+    }));
+
+    const option = do {
+      const matchedOption = options.find(
+        o => o.value === this.props.value.currencyCode
+      );
+      if (matchedOption) matchedOption;
+      // ensure an option is found, even when the currencies don't include
+      // the money value's currencyCode
+      else if (this.props.value.currencyCode.trim() !== '')
+        ({
+          label: this.props.value.currencyCode,
+          value: this.props.value.currencyCode,
+        });
+      else null;
+    };
     return (
       <Contraints.Horizontal constraint={this.props.horizontalConstraint}>
         <div
-          ref={this.containerRef}
           className={styles['field-container']}
-          onBlur={event => {
-            // ensures that both fields are marked as touched when one of them
-            // is blurred
-            if (
-              typeof this.props.onBlur === 'function' &&
-              !this.containerRef.current.contains(event.relatedTarget)
-            ) {
-              this.props.onBlur({
-                target: {
-                  id: MoneyInput.getCurrencyDropdownId(this.props.id),
-                  name: getCurrencyDropdownName(this.props.name),
-                },
-              });
-              this.props.onBlur({
-                target: {
-                  id: MoneyInput.getAmountInputId(this.props.id),
-                  name: getAmountInputName(this.props.name),
-                },
-              });
-            }
-          }}
+          data-testid="money-input-container"
         >
-          {this.props.currencies.length > 0 ? (
-            <CurrencyDropdown
-              id={MoneyInput.getCurrencyDropdownId(this.props.id)}
-              name={getCurrencyDropdownName(this.props.name)}
-              currencies={this.props.currencies}
-              currencyCode={this.props.value.currencyCode}
-              onChange={this.handleCurrencyChange}
-              isDisabled={this.props.isDisabled}
-              hasError={this.props.hasError}
-              hasWarning={this.props.hasWarning}
-            />
-          ) : (
-            <div className={styles['currency-label']}>
-              <div className={styles['currency-wrapper']}>
-                <Currency
-                  id={MoneyInput.getCurrencyDropdownId(this.props.id)}
-                  isDisabled={this.props.isDisabled}
-                  currency={this.props.value.currencyCode}
-                />
-              </div>
-            </div>
-          )}
+          <Select
+            inputId={MoneyInput.getCurrencyDropdownId(this.props.id)}
+            name={getCurrencyDropdownName(this.props.name)}
+            value={option}
+            isDisabled={this.props.isDisabled || hasNoCurrencies}
+            isSearchable={false}
+            components={{ DropdownIndicator, ClearIndicator }}
+            options={options}
+            placeholder=""
```

Comment:
The placeholder is empty because is the currency selector and we don't want anything there right?